### PR TITLE
CORE-14150. PcMemGetBiosMemoryMap(): Add a default handling of entry with "EA.ErrorLog = 1"

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Improve/Add FreeLdr logs.

JIRA issues: [CORE-13332](https://jira.reactos.org/browse/CORE-13332) and [CORE-14150](https://jira.reactos.org/browse/CORE-14150)

## Proposed changes

- pcmem.c: Improve output strings. No functional changes. CORE-13332
- PcMemGetBiosMemoryMap(): Add a default handling of entry with "EA.ErrorLog = 1". CORE-14150

### CODE REVIEW

**Comment by ThFabba** [comment]
So the point of this is to add a debug print that nobody will ever see?

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 1]
```diff
@@ -2,7 +2,7 @@
  *  FreeLoader
  *
  * Copyright ... ... (See below.)
- * Copyright 2017 Serge Gautherie
+ * Copyright 2017 Serge Gautherie <reactos-git_serge_171003@gautherie.fr>
```
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: No one really cares about emails actually, because they are subject to change at any time.

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 2]
```diff
@@ -224,8 +228,9 @@ PcMemCheckUsableMemorySize(VOID)
             __FILE__,
             __LINE__,
             "The BIOS reported a usable memory range up to 0x%lx, which is too small!\n"
-            "Required size is 0x%lx\n\n"
-            "If you see this, please report to the ReactOS team!",
+             "Required size is 0x%lx\n"
+             "\n"
```
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: Please keep the original.

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 3]
```diff
@@ -350,7 +355,7 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
 
         if (PcBiosMemoryMap[PcBiosMapCount].Length == 0)
         {
-            TRACE("Discard empty range. (would-be-PcBiosMapCount = %lu, BaseAddress = %lu, Length = 0)\n",
+            TRACE("Discard empty range. (would-be-PcBiosMapCount = %lu, BaseAddress = 0x%llx, Length = 0)\n",
```
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: Is this change needed to print pointers (of type PVOID) ? If so, why not using 0x%p ?

Comment (ThFabba) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: BaseAddress is ULONGLONG so this is right.

File: boot/freeldr/freeldr/arch/i386/pcmem.c (hunk 1 above)
Comment (ColinFinck) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Not true. Our standardized license headers in the Coding Style document on the Wiki request E-Mail addresses.
And most of them live longer than you may assume. I needed E-Mail addresses from all developers for the Git migration, and it turned out that many old ones of former developers are still used on GitHub today.

Comment (HBelusca) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Ah, good to know :)

File: boot/freeldr/freeldr/arch/i386/pcmem.c (hunk 2 above)
Comment (SergeGautherie) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Reverted.

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 4]
```diff
@@ -611,7 +626,7 @@ PcMemGetMemoryMap(ULONG *MemoryMapSize)
         }
     }
 
-    TRACE("DiskReadBuffer=%p, DiskReadBufferSize=%lx\n",
+    TRACE("DiskReadBuffer=%p, DiskReadBufferSize=0x%lx\n",
```
Comment (SergeGautherie) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: Then, do you want an added `0x` to `%p` here? (`0x%p` is used in the tree, but not much compared to `%p`...)

File: boot/freeldr/freeldr/arch/i386/pcmem.c [hunk 5]
```diff
@@ -348,9 +353,21 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
         TRACE("Type: 0x%lx\n", PcBiosMemoryMap[PcBiosMapCount].Type);
         TRACE("ExtendedAttributesAsULONG: 0x%08lx\n", PcBiosMemoryMap[PcBiosMapCount].ExtendedAttributesAsULONG);
 
+        if (PcBiosMemoryMap[PcBiosMapCount].ExtendedAttributes.ErrorLog == 1)
+        {
+            FIXME("EA.ErrorLog = 1. Please report this to CORE-14150. "
```
Comment (SergeGautherie) [inline boot/freeldr/freeldr/arch/i386/pcmem.c]: > So the point of this is to add a debug print that nobody will ever see?

Why would this `FIXME()` never be seen, precisely?
Hack-test: It works fine on VPC 2004, though the latter does not return EAs...

File: boot/freeldr/freeldr/arch/i386/pcmem.c (hunk 4 above)
Comment (HBelusca) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: I personally always put "0x" when I know I'm going to output hex.
Because if you see eg.
"DiskReadBufferSize = 3123", do you know at first sight whether this is in decimal or in hex?
When using the prefix you don't need to know what the code is doing to know it's hex.

Comment (SergeGautherie) [reply boot/freeldr/freeldr/arch/i386/pcmem.c]: Added.

**Comment by SergeGautherie** [comment]
I would like this PR to be committed before 0.4.8 branches.

**Comment by ROCKNROLLKID** [comment]
@gedmurphy You can look into merging this one, too, since you are so eager, haha.

**Comment by SergeGautherie** [comment]
Would it ease things somewhat if I move the CORE-13332 commit to a separate PR?
